include/swift/AST/DiagnosticsSema.def

```diff
@@ -1294,8 +1294,9 @@ ERROR(nominal_type_not_attribute,none,
 
 ERROR(mutating_invalid_global_scope,none, "%0 is only valid on methods",
       (SelfAccessKind))
-ERROR(mutating_invalid_classes,none, "%0 isn't valid on methods in "
-      "classes or class-bound protocols", (SelfAccessKind))
+ERROR(mutating_invalid_classes,none, "%0 isn't valid on %1 in " 
+      "%select{classes|class-bound protocols}2",
```

Comment:
One small remark - since diagnostic is singular based on `DescriptiveDeclKind` we need to either singularize `classes` and `protocols` or pluralize "kind" e.g. in modified tests:

> 'nonmutating' isn't valid on instance method in class-bound protocols

vs.

`'nonmutating' isn't valid on instance method in class-bound protocol`

or 

`'nonmutating' isn't valid on instance methods in class-bound protocols`

I think consistent pluralization reads better than a mix.

Reply:
Also please don't forget to update `en.yaml`

Reply:
Done!